File: extract_pdf.js
const fs = require('fs');
const pdf = require('pdf-parse');

const pdfPath = 'C:/Users/chean/OneDrive/Desktop/the-complete-collection.pdf';
const txtPath = 'C:/Users/chean/OneDrive/Desktop/the-complete-collection.txt';

console.log('Reading PDF from: ' + pdfPath);
try {
    let dataBuffer = fs.readFileSync(pdfPath);
    console.log('PDF read successfully, starting extraction...');

    pdf(dataBuffer).then(function(data) {
        console.log('Number of pages: ' + data.numpages);
        fs.writeFileSync(txtPath, data.text, 'utf8');
        console.log('Extraction complete! Text saved to ' + txtPath);
    }).catch(function(error) {
        console.error('Error parsing PDF:', error);
    });
} catch (error) {
    console.error('Error reading file:', error);
}

File: extract_pdf.py
import sys
import subprocess

try:
    import fitz  # PyMuPDF
except ImportError:
    print("Installing PyMuPDF...")
    subprocess.check_call([sys.executable, "-m", "pip", "install", "PyMuPDF"])
    import fitz

def extract_text(pdf_path, txt_path):
    print(f"Opening {pdf_path}...")
    try:
        doc = fitz.open(pdf_path)
        text = ""
        for i, page in enumerate(doc):
            text += page.get_text()
        
        print(f"Writing to {txt_path}...")
        with open(txt_path, "w", encoding="utf-8") as f:
            f.write(text)
        print("Extraction complete! Saved to", txt_path)
    except Exception as e:
        print(f"An error occurred: {e}")

if __name__ == "__main__":
    pdf_path = r"C:\Users\chean\OneDrive\Desktop\the-complete-collection.pdf"
    txt_path = r"C:\Users\chean\OneDrive\Desktop\the-complete-collection.txt"
    extract_text(pdf_path, txt_path)
